#! /usr/bin/env python
# 
# The path integral parameters for controlling an omni-wheeled robot, with
# the objective of achieving the maximum speed possible around a track
#

PACKAGE='autorally_control'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name       Type      Level Description     Default Min   Max
gen.add("max_wheel_speed", double_t, 0, "Maximum permissable wheel speed", 0.0, 0.0, 5.0)
gen.add("desired_speed", double_t, 0, "Speed Target for the MPPI controller", 0.5, 0.0, 3.0)
gen.add("speed_coefficient", double_t, 0, "Weight for acheiving target velocity", 4.25, 0.0, 20.0)
gen.add("track_coefficient", double_t, 0, "Weight for staying on the track", 200.0, 0, 500.0)
gen.add("crash_coefficient", double_t, 0, "Penalty for crashing", 10000, 0, 20000)
gen.add("track_slop", double_t, 0, "Value for clipping track cost to zero.", 0, 0, .75)

exit(gen.generate(PACKAGE, "PathIntegral", "OmniWheelRobotPathIntegralParams"))
